airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -171,14 +190,21 @@ public AirbyteCatalog discover(JsonNode config) throws Exception {
   @Override
   public AutoCloseableIterator<AirbyteMessage> read(final JsonNode config, final ConfiguredAirbyteCatalog catalog, final JsonNode state)
       throws Exception {
-    final SSHTunnel tunnel = SSHTunnel.getInstance(config);
-    tunnel.openTunnelIfRequested();
-    // this check is used to ensure that have the pgoutput slot available so Debezium won't attempt to
-    // create it.
-    final AirbyteConnectionStatus check = super.check(config); // hack to get around double ssh problem.
-
-    if (check.getStatus().equals(AirbyteConnectionStatus.Status.FAILED)) {
-      throw new RuntimeException("Unable establish a connection: " + check.getMessage());
+    final SshTunnel tunnel = SshTunnel.getInstance(config);
```

Comment:
You're getting a tunnel instance here but I don't see a call to open the tunnel.  Is that a coding mistake, or is the call happening farther in?

Reply:
i changed the structure of the class so that when the tunnel class is instantiated it immediately creates thea ctually ssh tunnel.